modulemd/v2/tests/test-modulemd-merger.c

```diff
@@ -100,6 +104,47 @@ merger_test_deduplicate (CommonMmdTestFixture *fixture,
   g_assert_cmpstr (baseline, ==, deduplicated);
 }
 
+static void
+merger_test_merger_with_real_world_data (CommonMmdTestFixture *fixture,
+                         gconstpointer user_data)
+{
+  g_autoptr (ModulemdModuleIndex) fedora_index = NULL;
+  g_autoptr (ModulemdModuleIndex) updates_index = NULL;
+  g_autoptr (ModulemdModuleIndex) merged_index = NULL;
+  g_autoptr (ModulemdModuleIndexMerger) merger = NULL;
+  g_autofree gchar *yaml_path = NULL;
+  g_autoptr (GPtrArray) failures = NULL;
+  g_autoptr (GError) error = NULL;
+  
+  yaml_path =
+    g_strdup_printf ("%s/modulemd/v2/tests/test_data/long-valid.yaml",
+                     g_getenv ("MESON_SOURCE_ROOT"));
+  
+  
+  fedora_index = modulemd_module_index_new ();
+  
+  yaml_path =
+    g_strdup_printf ("%s/modulemd/v2/tests/test_data/long-valid.yaml",
+                     g_getenv ("MESON_SOURCE_ROOT"));
+
+  updates_index = modulemd_module_index_new ();
+  
+  g_assert_nonnull (yaml_path);
+  
+  merger = modulemd_module_index_merger_new ();
+  g_assert_nonnull (merger);
+  
+  modulemd_module_index_merger_associate_index (merger, fedora_index, 0);
+  modulemd_module_index_merger_associate_index (merger, updates_index, 0);
+  
+  merged_index = modulemd_module_index_merger_resolve (merger, &error);
+  g_assert_nonnull (merged_index);
+  g_assert_null (error);
+  g_clear_object(&merged_index);
```

Comment:
It's not enough here to verify that the `merged_index` is not NULL. You need to actually validate that the merge results match what you expect.

You can either dump the YAML to a string and comparing it to a known plaintext or pick a few objects that you know should be updated during the merge and do a deep-comparison of them to see if they match the expected results.

Reply:
I have added cmpstr with a baseline and newline (sumped yaml string). Is it enough. It is in the same way the deduplicated test was written.

Reply:
That only worked for the deduplication test because the two indices were in fact exactly the same. If you are doing a proper merge here with `f29` and `f29-updates`, the result is expected to be entirely different from either of the inputs.

Reply:
Can you please elaborate. I am getting confused on what actually I need to do. I have added f29 as well.

Reply:
It may not be a good idea to start with such large data sets for this test. It may be a better idea to start small: maybe create two different ModuleStream YAML files and verify that a merge stores both of them to the output.

We can work up to replacements and complicated interactions.

Reply:
When I searched about ModuleStream, this is what I found :" STREAMS enables you to create modules to provide standard data communications services and then manipulate the modules on a stream. From the application level, modules can be dynamically selected and interconnected. No kernel programming, compiling, and link editing are required to create the interconnection". But I don't know exactly how to write statements to create a ModuleStream YAML files. Can you please guide me?

Reply:
Also can you please tell me if I am going in the right direction or not?

Reply:
You are not. I'm not sure where you got that message from, but it's entirely unrelated to Fedora Modularity. (It looks like it may have to do with Linux kernel modules).

ModuleStream YAML is described by https://github.com/fedora-modularity/libmodulemd/blob/master/spec.v2.yaml and many examples are provided in the [`modulemd/v2/tests/test_data](https://github.com/fedora-modularity/libmodulemd/tree/master/modulemd/v2/tests/test_data) directory in the codebase.

The `f29.yaml` contains MANY such documents inside a single YAML file. An example of a single ModuleStream document would be [`dependson_v2.yaml`](https://github.com/fedora-modularity/libmodulemd/blob/master/modulemd/v2/tests/test_data/dependson_v2.yaml)

What I'm asking is for you to create another example module stream document similar to [`dependson_v2.yaml`](https://github.com/fedora-modularity/libmodulemd/blob/master/modulemd/v2/tests/test_data/dependson_v2.yaml), load each of them into an index, merge the two indices and then verify that the output is as expected.